packages/parser/src/parser.ts

```diff
@@ -160,14 +160,11 @@ function parseForESLint(
         analyzeOptions.jsxFragmentName,
       );
     }
-    if (compilerOptions.emitDecoratorMetadata === true) {
-      emitDecoratorMetadata = true;
-    }
   }
 
-  if (emitDecoratorMetadata) {
-    analyzeOptions.emitDecoratorMetadata = true;
-  }
+  // if not defined - override from the parserOptions
+  services.emitDecoratorMetadata ??= options.emitDecoratorMetadata === true;
+  services.experimentalDecorators ??= options.experimentalDecorators === true;
```

Comment:
[Refactor] `createParserServices.ts` also has defaulting logic around these two options, along with a now-contradictory comment:

https://github.com/typescript-eslint/typescript-eslint/blob/1c1f9f46b05ff8cf777c6555994b30c0e47da8ef/packages/typescript-estree/src/createParserServices.ts#L27-L29

How about passing through the `options` through `parseAndGenerateServices` to `createParserServices`? That way the defaulting logic can all be in one place.

```ts
  return {
    program,
    emitDecoratorMetadata:
      compilerOptions.emitDecoratorMetadata ??
      originalOptions.emitDecoratorMetadata === true,
    experimentalDecorators:
      compilerOptions.experimentalDecorators ??
      originalOptions.experimentalDecorators === true,
```

Aside: I found the naming of the options variables in `parser.ts` to be a little confusing to read through. There's an `options` variable of type `ParserOptions` and a `parserOptions` variable of type `TSESTreeOptions`... oy.

Reply:
The reason I didn't do that is because I didn't want the ESLint parser options to bleed into the typescript-estree parser options.

Ultimately `parserOptions.emitDecoratorMetadata` and `parserOptions.experimentalDecorators` are not used at all within `typescript-estree` - so it seems wrong to pass them all the way through into that package.

I think it's better to keep the logic isolated to `parser` so it only impacts true eslint parser.